protocols/rendezvous/src/codec.rs

```diff
@@ -236,6 +243,74 @@ impl Decoder for RendezvousCodec {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct Codec {}
+
+#[async_trait]
+impl libp2p_request_response::Codec for Codec {
+    type Protocol = StreamProtocol;
+    type Request = Message;
+    type Response = Message;
+
+    async fn read_request<T>(&mut self, _: &Self::Protocol, io: &mut T) -> io::Result<Self::Request>
+    where
+        T: AsyncRead + Unpin + Send,
+    {
+        let mut frame = FramedRead::new(io, RendezvousCodec::default());
+        if let Some(result) = frame.next().await {
+            return Ok(result?);
+        }
+
+        Err(io::ErrorKind::InvalidInput.into())
```

Comment:
You could use the `.ok_or` fn on the `Option` returned from `next` to express this using combinators.

Reply:
<img width="806" alt="image" src="https://github.com/libp2p/rust-libp2p/assets/25565200/02327073-0856-4edb-b0fa-414969e41bd4">

If I understand you right, this way doesn't look elegant.
I just inline an extra variable.

Reply:
The thing with `ok_or` is that you can now `?` it and don't have to write separate return statements:

```rust
let mut framed = FramedRead::new(io, RendezvousCodec::default());

let message = frame.next().await.ok_or(io::ErrorKind::InvalidInput)?;

// ...
```

Reply:
Ok, I got it

Reply:
@thomaseizinger 
```
let message = framed.next().await.ok_or(io::ErrorKind::InvalidInput)?
```
^ the trait `From<std::result::Result<_, _>>` is not implemented for `std::io::Error`